src/shogun/labels/BinaryLabels.cpp

```diff
@@ -6,17 +6,22 @@
  *          Chiyuan Zhang, Olivier NGuyen, Thoralf Klein
  */
 
-#include <shogun/labels/DenseLabels.h>
+#include <shogun/base/range.h>
 #include <shogun/labels/BinaryLabels.h>
-#include <shogun/mathematics/Statistics.h>
+#include <shogun/labels/DenseLabels.h>
 #include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Statistics.h>
 
 using namespace shogun;
 
 CBinaryLabels::CBinaryLabels() : CDenseLabels()
 {
 }
 
+CBinaryLabels::CBinaryLabels(const CBinaryLabels& orig) : CDenseLabels(orig)
```

Comment:
I gave labels copy constructors to the sake of converting them, see `as_binary` below

Reply:
why do we need a copy? i mean it's over the same datastruct no? do we really want the `as_binary` be a copy? dont we wanna use c++11's feature of std::move and rvalue?

Reply:
Yeah I thought about this. The reason why I made it a new instance was the fact that in the `as_binary` below, all other cases return a new instance, so to be consistent.
And also then the `as_binary` method can be const. On the other hand, `as_binary` could return const pointer ...
If you think it is better to just cast, I can do that. Not sure I get what you mean with the move and rvalue

Reply:
yep!

Reply:
so then use rvalue/move instead of this pattern.

Reply:
how do you want to deal with converting say multiclass labels (that only contain two classes) to binary? For this, the actual vector needs to be changed, so a new instance is needed.

Reply:
For converting the same datastruct, I see the point

Reply:
I think it is fine to copy BinaryLabels most of the time. What should be rather copy-on-write is the data beneath.

Reply:
I would actually prefer if this thing gave you a read-only labels object (we wanna have immutable labels anyways)
Note that memory is not copied here.
Also, after reading on it, I think the move constructor stuff doesnt really help here